operator-common/src/main/java/io/strimzi/operator/common/operator/resource/AbstractNamespacedResourceOperator.java

```diff
@@ -97,27 +112,42 @@ public Future<ReconcileResult<T>> reconcile(Reconciliation reconciliation, Strin
             return Future.failedFuture("Given name " + name + " incompatible with desired name " + desired.getMetadata().getName());
         }
 
-        return getAsync(namespace, name)
-                .compose(current -> {
-                    if (desired != null) {
-                        if (current == null) {
-                            LOGGER.debugCr(reconciliation, "{} {}/{} does not exist, creating it", resourceKind, namespace, name);
-                            return internalCreate(reconciliation, namespace, name, desired);
-                        } else {
-                            LOGGER.debugCr(reconciliation, "{} {}/{} already exists, updating it", resourceKind, namespace, name);
-                            return internalUpdate(reconciliation, namespace, name, current, desired);
-                        }
-                    } else {
-                        if (current != null) {
-                            // Deletion is desired
-                            LOGGER.debugCr(reconciliation, "{} {}/{} exist, deleting it", resourceKind, namespace, name);
-                            return internalDelete(reconciliation, namespace, name);
+        if (useServerSideApply) {
+            if (desired != null) {
+                if (desired.getMetadata().getManagedFields() != null && !desired.getMetadata().getManagedFields().isEmpty()) {
+                    LOGGER.debugCr(reconciliation, "Deleting managedFields from request before pathing resource {} {}/{}", resourceKind, namespace, name);
+                    desired.getMetadata().setManagedFields(null);
```

Comment:
Why is this needed? Maybe some comments would be useful?

Reply:
We don't need to update managed fields (the internal metadata about SSA / field ownership) at all. It caused problems with `StrimziPodSet` where whole `StrimziPodSet` object was replaced. 
I think it was `io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java:527` but I am not sure :)